conans/test/functional/command/profile_test.py

```diff
@@ -14,140 +13,152 @@
-            profiles.append("link" + os.path.sep + "profile7")
-
-        # Make sure local folder doesn't interact with profiles
-        os.mkdir(os.path.join(client.current_folder, "profile3"))
-        client.run("profile list")
-        for p in profiles:
-            assert p in client.out
-
-        # Test profile list json file
-        # FIXME: Json cannot be captured from tests?
-        # client.run("profile list --format=json > profiles_list.json")
-        # json_content = client.load("profiles_list.json")
-        # json_obj = json.loads(json_content)
-        # self.assertEqual(list, type(json_obj))
-        # self.assertEqual(profiles, json_obj)
-
-    def test_show(self):
-        client = TestClient()
-        profile1 = textwrap.dedent("""
-            [settings]
-            os=Windows
-            [options]
-            MyOption=32
-            [conf]
-            tools.build:jobs=20
-            """)
-        save(os.path.join(client.cache.profiles_path, "profile1"), profile1)
-
-        client.run("profile show -pr=profile1")
-        self.assertIn("[settings]\nos=Windows", client.out)
-        self.assertIn("MyOption=32", client.out)
-        # self.assertIn("CC=/path/tomy/gcc_build", client.out)
-        # self.assertIn("CXX=/path/tomy/g++_build", client.out)
-        # self.assertIn("package:VAR=value", client.out)
-        self.assertIn("tools.build:jobs=20", client.out)
-
-    def test_missing_subarguments(self):
-        client = TestClient()
-        client.run("profile", assert_error=True)
-        self.assertIn("ERROR: Exiting with code: 2", client.out)
-
-
-class DetectCompilersTest(unittest.TestCase):
-    def test_detect_default_compilers(self):
-        platform_default_compilers = {
-            "Linux": "gcc",
-            "Darwin": "apple-clang",
-            "Windows": "msvc"
-        }
-
-        result = detect_defaults_settings()
-        # result is a list of tuples (name, value) so converting it to dict
-        result = dict(result)
-        platform_compiler = platform_default_compilers.get(platform.system(), None)
-        if platform_compiler is not None:
-            self.assertEqual(result.get("compiler", None), platform_compiler)
-
-    @pytest.mark.tool("gcc")
-    @pytest.mark.skipif(platform.system() != "Darwin", reason="only OSX test")
-    def test_detect_default_in_mac_os_using_gcc_as_default(self):
-        """
-        Test if gcc in Mac OS X is using apple-clang as frontend
-        """
-        # See: https://github.com/conan-io/conan/issues/2231
-        output = check_output_runner("gcc --version", stderr=subprocess.STDOUT)
-
-        if "clang" not in output:
-            # Not test scenario gcc should display clang in output
-            # see: https://stackoverflow.com/questions/19535422/os-x-10-9-gcc-links-to-clang
-            raise Exception("Apple gcc doesn't point to clang with gcc frontend anymore!")
-
-        output = RedirectedTestOutput()  # Initialize each command
+
```

Comment:
Moved the file away from unittest - Maybe I should have left the classes be?

Reply:
A bit difficult to see the relevant changes that could be caused by the new feature, but not a big problem